src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable no-console */
+'use strict';
+
+const { expensesService } = require('../services/expenses.service.js');
+const { userService } = require('../services/user.service.js');
+
+const get = async(req, res) => {
+  const expenses = await expensesService.getAll(req.query);
+
+  res.send(expenses);
+};
+
+const getOne = async(req, res) => {
+  const { id } = req.params;
+
+  const expense = await expensesService.getById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expense);
+};
+
+const create = async(req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  if (!userId || !spentAt || !title || amount < 0 || !category || !note) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  try {
+    await userService.getById(userId);
+  } catch (error) {
+    console.log('User not found');
+    res.sendStatus(400);
+
+    return;
+  }
+
+  if (!(new Date(spentAt) instanceof Date)
+    || typeof title !== 'string'
+    || typeof amount !== 'number'
+    || typeof category !== 'string'
+    || typeof note !== 'string'
+  ) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const expense = await expensesService.create(req.body);
+
+  res.statusCode = 201;
+
+  res.send(expense);
+};
```

Comment:
to much logic in controller. it should be responsible only for taking data from request and put it in service (all validation, business logic, etc. should be handled  by service), like this:
```js
const create (req,res) => {
  try {
     const expense = await expensesService.create(req.body);
     res.statusCode = 201;
  } catch (e) {
    console.error(`error creating expense: ${e.message}`)
  }
}
```

Reply:
please fix it everywhere